lib/reports/BackupReportManager.js

```diff
@@ -4,85 +4,131 @@ const _ = require('lodash');
+                //check if more than one entries present in jobrundetails on delete day
+                return getJobRunDetailCount(moment.utc(endTime).startOf('day').toDate(), endTime)
+                  .then(count => {
+                    if (!count) {
+                      logger.info('instance deleted before backup was taken on ', endTime);
+                      endTime = moment.utc(endTime).subtract(1, 'days').endOf('day').toDate();
+                    }
+                  });
+              }
+            })
+          ])
+          .then(() => {
+            //iterate through list of days and for each day count successful backups
+            for (; currentDay < endTime; currentDay = moment(currentDay).add(1, 'days').toDate()) {
+              logger.info('checking for backups on ', currentDay);
+              let day_end = endTime < moment(currentDay).add(1, 'days').toDate() ? endTime : moment(currentDay).add(1, 'days').toDate();
+              //iterator to last entry on the day
+              let endIndex = startIndex;
+              while (
+                endIndex < resultBackups.length &&
+                resultBackups[endIndex].metric === config.monitoring.success_metric &&
+                resultBackups[endIndex].createdAt < day_end
+              ) {
+                endIndex++;
+              }
+              backupSuccessCount += (endIndex - startIndex);
+              if (startIndex === endIndex) {
+                logger.info('No backup day', currentDay);
+                noBackupDays.push(currentDay);
+              }
+              startIndex = endIndex;
+            }
+            // check consecutive_backup_failure_sla_count failure count
+            const maxAllowedNoBackupDaysInRow = config.backup.consecutive_backup_failure_sla_count;
+            let windowStartIndex = 0; // start index of consecutive no backup days window
+            let windowEndIndex = 0; // end index of consecutive no backup days window
+            while (windowEndIndex < noBackupDays.length) {
+              while (
+                windowEndIndex + 1 < noBackupDays.length &&
+                noBackupDays[windowEndIndex].valueOf() === moment(noBackupDays[windowEndIndex + 1]).subtract(1, 'days').toDate().valueOf()
+              ) {
+                windowEndIndex++;
+              }
+              consecutiveBackupFailureCount += (windowEndIndex - windowStartIndex + 1) >= maxAllowedNoBackupDaysInRow ? (windowEndIndex - windowStartIndex + 2 - maxAllowedNoBackupDaysInRow) : 0;
+              windowEndIndex++;
+              windowStartIndex = windowEndIndex;
+            }
+            const summary = {
+              noBackupDays: noBackupDays,
+              backupsTriggerred: triggeredBackupCount,
+              backupsSucceeded: backupSuccessCount,
+              backupFailed: (resultBackups.length - backupSuccessCount),
+              failedBackupGuids: resultBackups.slice(backupSuccessCount),
```

Comment:
One comment not with regards to here, but can you also tweak the script `generate-backup-report` to add another column `fail_reason` along with `failed_backup_guids` something like
```
failed_backup_reason=$(echo "$backup_summary" |
                            jq '.failedBackupGuids[] | "\(.response) \(.createdAt)"' |
                            tr '\n' ' ' |
                            sed -e 's/,/;/g'
                          )
```
This is a minor fix, but it will certainly make trouble shooting almost unnecessary. I know the column would be really huge, but that anyways would be word wrapped & only interested folks would expand and look at it. What say? Atleast as an analyzer of backup report this field certainly makes the job much easier.

Reply:
Yes, that was the idea.

Reply:
rename `failedBackupGuids` to `failedBackups`